internal/diffmapper/diffmapper.go

```
@@ -0,0 +1,86 @@
+// Package diffmapper maps git diff with original file line number
+package diffmapper
+
+import (
+	"bufio"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type ChangedLines []ChangedLine
+
+type ChangedLine struct {
+	LineNum int
+	Content string
+}
+
+var latestChanges ChangedLines
+
+func GetMap(diff string) ChangedLines {
+	hunkStarted := false
+	var changes []ChangedLine
+	scanner := bufio.NewScanner(strings.NewReader(diff))
+
+	var newLineNum int
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.HasPrefix(line, "@@") {
+			hunkStarted = true
+			// Parse hunk header: @@ -a,b +c,d @@
+			parts := strings.Split(line, " ")
+			newHunk := parts[2] // "+c,d"
+			hunkStart := strings.Split(strings.TrimPrefix(newHunk, "+"), ",")
+			newLineNum, _ = strconv.Atoi(hunkStart[0])
```

Error handling is missing when parsing the hunk header. The strconv.Atoi() call ignores errors, which could lead to unexpected behavior if the diff format
is invalid.